src/Makefile.am

```diff
@@ -808,8 +808,7 @@ tests_testMem_LDFLAGS = $(LIBADD_DL)
 
```

Comment:
> PR description: Most unit test classes are trivial, it's not necessary to have a header file and an implementation file for them, becasuse the definition is never reused

The opening assertion -- "most test classes are trivial" -- is correct but irrelevant. It is the last assertion -- "never reused" -- that actually supports your (correct) decision to remove/merge those headers. Also, those test classes are not meant or not designed to be reused across translation units: Some effort is required to make some of them reusable and we should not false-advertise otherwise.

I have adjusted the PR description to remove the irrelevant assertion and add another argument in support of this PR. I also fixed spelling. Please check and adjust further as needed, of course.